internal/cmd/tls.go

```diff
@@ -0,0 +1,255 @@
+
+	"github.com/cdr/grip"
+	"github.com/spf13/cobra"
+	"golang.org/x/xerrors"
+
+	"cdr.dev/coder-cli/coder-sdk"
+	"cdr.dev/coder-cli/internal/coderutil"
+	"cdr.dev/coder-cli/pkg/clog"
+	"cdr.dev/coder-cli/pkg/tablewriter"
+)
+
+type ErrMissingRequiredParameter struct {
+	Name string
+}
+
+func (e ErrMissingRequiredParameter) Error() string {
+	return fmt.Sprintf("Missing required parameter '%s'", e.Name)
+}
+
+func tlsCmd() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "tls",
+		Short: "Manage Coder TLS configuration",
+		Long:  "Manage Coder TLS configuration via self-signed certificates, custom certificates, or Let's Encrypt certificates",
+	}
+
+	cmd.AddCommand(
+		tlsSelfSign(),
+		tlsCustom(),
+		tlsACME(),
+		tlsDisable(),
+	)
+	return cmd
+}
+
+func tlsSelfSign() *cobra.Command {
+	var (
+		hosts []string
+	)
+	cmd := &cobra.Command{
+		Use:     "self-sign",
+		Short:   "Generate self-signed certificate for Coder",
+		Long:    "Generate self-signed certificates for Coder. Self-signed certificates are automatically renewed by Coder",
+		Example: "tls self-sign --hosts a.example.com --hosts b.example.com --hosts c.example.com",
+		RunE: func(cmd *cobra.Command, args []string) error {
+			ctx := cmd.Context()
+			client, err := newClient(ctx, true)
+			if err != nil {
+				return err
+			}
+
+			if len(hosts) == 0 {
+				return ErrMissingRequiredParameter{"hosts"}
+			}
+
+			resp, err := client.GenerateSelfSignedCertificate(ctx, coder.SelfSignedCertificateRequest{
+				Hosts: hosts,
+			})
+			if err != nil {
+				return err
+			}
+
+			if !resp.Success {
+				return xerrors.New("Failed to generate self-signed certificate")
+			}
+
+			clog.LogSuccess("Generated self-signed certificate")
+			return nil
+		},
+	}
+
+	cmd.Flags().StringArrayVar(&hosts, "hosts", []string{}, "Hostnames and/or IPs to generate self-signed certificate for")
+
+	return cmd
+}
+
+func tlsCustom() *cobra.Command {
+	var (
+		certFilepath string
+		keyFilepath  string
+	)
+
+	cmd := &cobra.Command{
+		Use:     "custom",
+		Short:   "Upload custom PEM encoded certificate and key files",
+		Example: "tls custom --cert /path/to/cert.pem --key /path/to/key.pem",
+		RunE: func(cmd *cobra.Command, args []string) error {
+			ctx := cmd.Context()
+			client, err := newClient(ctx, true)
+			if err != nil {
+				return err
+			}
+
+			cert, err := os.ReadFile(certFilepath)
+			if err != nil {
+				return err
+			}
+
+			key, err := os.ReadFile(keyFilepath)
+			if err != nil {
+				return err
+			}
+
+			req := coder.UploadCertificateRequest{
+				Certificate: cert,
+				PrivateKey:  key,
+			}
+
+			resp, err := client.UploadCustomCertificate(ctx, req)
+			if err != nil {
+				return err
+			}
+
+			if !resp.Success {
+				return xerrors.New("Failed to apply custom TLS certificate")
+			}
+
+			clog.LogSuccess("Uploaded custom TLS certificate")
+			return nil
+		},
+	}
+
+	cmd.Flags().StringVarP(&certFilepath, "cert", "", "", "Full path to a PEM encoded certificate file")
```

Comment:
The P-suffixed methods are for the short flags, and I think you can use StringVar() if there's no short form. From the [docs for pflag](https://pkg.go.dev/github.com/spf13/pflag):

>The pflag package also defines some new functions that are not in flag, that give one-letter shorthands for flags. You can use these by appending 'P' to the name of any function that defines a flag.